src/pid/limits.rs

```diff
@@ -0,0 +1,335 @@
+//! Process limits informations from `/proc/[pid]/limits`.
+
+use std::fs::File;
+use std::io::Result;
+use std::str::{self};
+use time::Duration;
+
+use libc::pid_t;
+use nom::{
+    IResult,
+    line_ending,
+    is_space
+};
+
+use parsers::{
+    map_result,
+    parse_isize,
+    parse_word,
+    read_to_end
+};
+
+/// Parses a limit value
+/// turns it into an isize for later processing
+named!(parse_limit_value<&[u8], isize>,
+    alt!(map!(tag_s!("unlimited"), |_| LIMITS_INFINITY) | parse_isize)
+);
+
+/// Parses a limit line
+named!(parse_limit_line<&[u8],(isize,isize,Option<LimitUnit>)>,
+    do_parse!(
+        // We don't care about the limit name
+        take_until!("  ") >>
+        // Take while we hit a space
+        take_while!(is_space) >>
+        // Parses soft limit
+        soft_limit: parse_limit_value >>
+        // Take while we hit a space
+        take_while!(is_space) >>
+        // Parses hard limit
+        hard_limit: parse_limit_value >>
+        // Take while we hit a space
+        take_while!(is_space) >>
+        // Parses the unit of the limit. The value may not be specified
+        unit: map!(opt!(parse_word), unit_types) >>
+        // Take while we hit a space
+        take_while!(is_space) >>
+        // End of the limit line
+        line_ending >>
+
+        ((soft_limit, hard_limit, unit))
+    )
+);
+
+/// A constant to represent the "unlimited" value
+pub const LIMITS_INFINITY: isize = -1;
+
+/// A limit unit Enum. Has a field for every unit we may encounter (according to Linux version)
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum LimitUnit{
+    Seconds,
+    Bytes,
+    Processes,
+    Files,
+    Locks,
+    Other(String),
+    Signals,
+    Us
+}
+
+/// A struct to hold limits and unit of the limit type
+/// A soft_limit or hard_limit equals to -1 means its "unlimited".
+#[derive(Debug, PartialEq, Eq)]
+pub struct Limit{
+    pub soft_limit: isize,
```

Comment:
Likewise, I'm not a fan of this struct.  I think it would be better to have two fields for each line in /proc/../limits, e.g. `max_cpu_time_soft` and `max_cpu_time_hard`.

`isize` does _not_ have sufficient range for representing the limit values.  File sizes should be in terms of `u64`, memory sizes and counts should be in terms of `usize`.  Additionally, each should be wrapped in an `Option` for representing unlimited (if unlimited is possible).